### PR TITLE
Update to post-challenge and new challenge

### PR DESCRIPTION
**Pull request by colbygk**
I was unfamiliar with working against Reddit's APIs, so please feel full freedom to reject or request alterations to this pull request.

What I found was:

- It appears that the praw api has evolved since post-challenges.py was created and I have updated post-challenges.py according to what I believe is the expected behavior.
- As a test, I have added in challenge 336, which post-challenges.py pulled from Reddit.
- In a future pull request, I would like to update the documentation to provide guidance on how to use praw and the setting up of a reddit api access key, as that information was not in a convenient single location.
  - I would also like to update the documentation about the reference to a post_challenges script that seems to no longer exist in the repo and another script referenced inside of post-challeneges.py that is also not there (send-data.sh)
- After we have discussed these updates, I will submit more pull requests that fit what you would prefer.
- The naming convention appears to have shifted and challenges appear in the repo that do not follow what is requested in CONTRIBUTING.md, i.e. #336 vs 0336 and so on.  Would be happy to rename and resubmit those to follow the naming convention

### CODE REVIEW

File: post-challenges.py [hunk 1]
```diff
@@ -27,11 +27,11 @@ def get_current_week(num_challenges=1):
...
 
 	r = praw.Reddit(user_agent="dailyprogrammer-challenges")
-	sub = r.get_subreddit("dailyprogrammer")
+	sub = r.subreddit("dailyprogrammer")
 
 	# retrieve generators for top posts
-	chals = sub.get_new(limit=num_challenges)
-	_chals = sub.get_new(limit=num_challenges)
+	chals = sub.new(limit=num_challenges)
+	_chals = sub.new(limit=num_challenges)
```
Comment (freddiev4) [inline post-challenges.py:34]: 👍  for update to match the current PRAW API

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+praw>=5.1.0
+pprint>=0.1
```
Comment (freddiev4) [inline requirements.txt]: 👍  for adding a `requirements.txt` file.

Can we move this to a `etc/` folder?

File: post-challenges.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def get_current_week(num_challenges=1):
 		logging.info("Created solutions directory")
 
 	logging.info("Started sending data script")
-	os.system("./send-data.sh")
+#	os.system("./send-data.sh")
```
Comment (freddiev4) [inline post-challenges.py]: I think we can just remove this line along with the log above it

**Review by freddiev4 — COMMENTED** [review]
Hey @colbygk awesome work. I have some feedback for you below.

In regards to the naming scheme, see if you can run `transform.py` which should end up renaming directories in the proper format.

**Comment by colbygk** [comment]
Hi @freddiev4 I have submitted an update to transform that I think will help it do a better job. I have also backfilled the challenges to fill the gap between 313 and this current week. I just noticed that there appears to be no 334 easy challenge (in the reddit itself)

File: transform.py [hunk 4]
```diff
@@ -19,57 +19,50 @@ def update_challenge_dir_name(oldName, difficulty):
...
-    else:
-        challengeName = ""
-
-    newTitle = newTitle + challengeName
-    return newTitle
+          
+    # Found a likely looking number
+    #   tokens now possess' the remaining title
+    #   a_token now has the original and possible multiple iteration number
+    #     such as 166b
```
Comment (freddiev4) [inline transform.py]: Just as a style thing, we should probably line up the first word of every line to be in the same column
```
    # Found a likely looking number
    #   tokens now possess' the remaining title
    #   a_token now has the original and possible multiple iteration number
    #     such as 166b
```

Changed to

```
    # Found a likely looking number
    # tokens now possess' the remaining title
    # a_token now has the original and possible multiple iteration number
    # such as 166b
``` 

For every set of comments

**Comment by freddiev4** [comment]
@colbygk just had one comment in regards to style. If you fix that I think we can go ahead and merge this 🙂 great work on refactoring `transform.py`!

**Comment by colbygk** [comment]
Okay, I think that's it. I did update the README.md that you may want to double check.

**Comment by colbygk** [comment]
Okay, fo'reals, I'm done.

File: README.md [hunk 5]
```diff
@@ -2,18 +2,39 @@
...
+The script can also be run via the command line by running `./post-challenges.py <number_of_challenges>`
+
+You may wish to run this in a temporary directory, to avoid adding folders directly to the top directory of the repo. Move the various levels of challenges into their respective end points.
+
+### Example full run
 
-**post-challenges.py runs on Python 3.x**
+  ```bash
+  $ cd DailyProgrammerChallenges
+  $ mkdir tmp
```
Comment (freddiev4) [inline README.md]: What are we using `tmp/` for here? From what you added above, it looks like we should `cd` into it.

**Review by freddiev4 — COMMENTED** [review]
Added one more comment + you should squash your commits 👍

File: README.md (hunk 5 above)
Comment (colbygk) [reply README.md]: Right you are.  Also added a bit about using praw.ini

**Comment by colbygk** [comment]
I am unfamiliar with commit squashing.  Recommendation on what to do to squash?

**Comment by colbygk** [comment]
Hm, I think I figured it out and squashed the last 5 commits into one.

**Comment by freddiev4** [comment]
@colbygk here's an official guide on [Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) in Git.

The short of it is that you can do:

`git rebase -i @~15` which will allow you to do an interactive rebase (`-i`), on the head of your branch (`@`), up until the last 15 commits (`~15`).

From there you'll see something like:
```
pick f7f3f6d changed my name a bit
pick 310154e updated README formatting and added blame
pick a5f4a0d added cat-file
```

And you can change everything below the first commit to:
```
pick f7f3f6d changed my name a bit
fixup 310154e updated README formatting and added blame
fixup a5f4a0d added cat-file
```

Which will remove the commit message of every commit but the first (which is 15 commits before the most recent one, the head of your branch), and also squash everything into just one commit

**Comment by colbygk** [comment]
Looks like since I used `squash` instead of `fixup` it melded instead of replacing.

**Comment by freddiev4** [comment]
After you squash, I think you'd need to run `git push -f`, as it looks like there are still 17 commits here 🙂 

image

**Comment by colbygk** [comment]
Well, now we're down to 11 commits. :)  Were you thinking that it should have more squashing?

**Comment by freddiev4** [comment]
Yeah usually we can squash many commits down into 2 or 3 commits.

Here I was thinking something like:

```
Update most recent challenges
Refactor post_challenges and transform.py
Update config, dev files, and docs
```

**Comment by colbygk** [comment]
Hmm... Okay... maybe done now? Sorry for the thrashing.

**Comment by freddiev4** [comment]
Looks good to me 👍

**Pull request merged**